### ISSUE TITLE
Better sanity check of commandline

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#143]
**Reported by latinsud on 18 Mar 2007 12:34 UTC**

As some people forget some arguments, it can happen this:
```
# aireplay-ng -1 -e MyHouse eth1
ioctl(SIOCGIFINDEX) failed: No such device
```

This is because "-e" has been accepted as the argument of "-1", and "MyHouse" been interpreted as the interface. Imho it's also wrong that aireplay-ng admits 2 extra arguments.

First suggestion is check return value of ALL sscanf in ALL programs:
```
- sscanf( optarg, "%d", &opt.a_delay );
- if( opt.a_delay < 0 )

+ ret=sscanf( optarg, "%d", &opt.a_delay );
+ if( ret!=1 || opt.a_delay < 0 )
```

Second suggestion is check extra argument count. 
In aireplay-ng, unless i'm wrong there must be exactly 1 extra argument (the injecting interface).
```
- if( argc - optind < 1 || argc - optind > 2 )

+ if( argc - optind != 1 )
```

In airodump-ng there's no extra "optind" argument count check, there should be.

**Comment by aircrack-ng** [comment]
**Modified by latinsud on 18 Mar 2007 12:34 UTC**

**Comment by aircrack-ng** [comment]
**Modified by latinsud on 18 Mar 2007 12:34 UTC**

**Comment by aircrack-ng** [comment]
**Modified by latinsud on 18 Mar 2007 12:34 UTC**

**Comment by aircrack-ng** [comment]
**Comment by hirte on 18 Mar 2007 12:34 UTC**

(In #230) Added more informations, better error messages and a --help switch (Closes: #143).

**Comment by aircrack-ng** [comment]
**Modified by misterx on 18 Mar 2007 12:34 UTC**
